### DISCUSSION TITLE
Is there a way to (or should i?) ignore kernel related vulns during a container scan?

### DISCUSSION THREAD

**Discussion by myarbrou-rh** [aquasecurity/trivy#3856] [Q&A] [answered]
Hey all

I have a few projects on Gitlab that run a Trivy scan provided by the [Gitlab templates](https://gitlab.com/gitlab-org/gitlab/-/blob/master/lib/gitlab/ci/templates/Jobs/Container-Scanning.gitlab-ci.yml). This scan currently runs on Gitlabs own runners which if i'm not mistaken means that the kernel related vulnerabilities like [CVE-2023-0461](https://nvd.nist.gov/vuln/detail/CVE-2023-0461) [CVE-2022-4379](https://nvd.nist.gov/vuln/detail/CVE-2022-4379) are issues with the host machine instead of the actual container.

a) Am I correct in assuming that vulnerabilities like this are not related to the container image that i'm building?
b) If so, is there a way to have trivy ignore vulnerabilities that are connected to the host machine that the scanned container is running on?

**Comment by AnaisUrlichs** [comment, marked as answer]
Hi there, sorry for the late reply, we actually got a similar question in [Slack](https://aquasecurity.slack.com/archives/C02KVB6AED9/p1688730931820899) 
"First, container images don't show the Linux kernel in the list of installed packages. Trivy doesn't detect kernel vulnerabilities anyway.
This issue is mainly relevant to host scanning like trivy rootfs or trivy vm."
https://github.com/aquasecurity/trivy/issues/3764#issuecomment-1457869338

I think this is an issue with GitLab rather than Trivy -- but why are you seeing the Vulnerabilities in the container image scan?

@itaysk @knqyf263  could you please provide further information on how this works

Regarding your second question -- have you tried trivyignore? https://aquasecurity.github.io/trivy/v0.43/docs/configuration/filtering/#by-finding-ids

**Reply by knqyf263** [reply]
As we discussed [here](https://github.com/aquasecurity/trivy/issues/1596), it is hard to say those vulnerabilities don't affect your container images since your container image probably has some packages installed from the `linux` package.

As @AnaisUrlichs shared, you can ignore them by specifying IDs.

And we might want to add a flag for ignoring kernel vulnerabilities even though it may miss some affected vulnerabilities. Kernel vulnerabilities indeed likely don't affect.

**Reply by RobinFrcd** [reply]
A flag to ignore kernel vulnerabilities that are not related to the docker image itself, but the host machine would indeed be a nice feature !

**Reply by AnaisUrlichs** [reply]
Hi @RobinFrcd, could you please create a separate feature request in the discussions for this and link to this thread? 🙂

**Reply by knqyf263** [reply]
https://github.com/aquasecurity/trivy/discussions/4816
